model/aberrations.js

```diff
@@ -64,66 +71,198 @@ exports.upsertAber = function(data, callback){
 	    err = Error("Did not return annotation ID")
 	}
 	if (err) {
-            console.log("Error upserting gene annotation: " + err);
-	    console.log("Debug: full query:", query.string)
-	    callback(err, null)
+            console.log("Error upserting gene aberration: " + err);
+	    callback(err, null);
+	    return
 	}
     }
 
-    // todo: test update case
-    // todo: transaction-ize w/ rollback (not necessary, just good to clean the annos table)
-    // TODO: may insert duplicates, fix this if we want something more interesting with genes
+    // TODO: we may insert duplicate mutations, fix this if we want something more interesting with genes
+
+    var now = new Date().toString().substring(0,15);
+    var acids=['?','?'], locus;
+    // parse the protein sequence change
+    if (data.protein_seq_change) {
+	acids = data.protein_seq_change.split(/\d+/);
+	locus = 0; 	
+	if (acids[0].length < data.protein_seq_change.length)
+	    locus = parseInt(data.protein_seq_change.substr(acids[0].length));
+	if (acids.length == 1)
+	    acids[1] = '?';
+    }
 
+    
     var aberInsertQuery = abers.insert(
-	abers.gene.value(data.gene),
-	abers.transcript.value(data.transcript),
-	abers.mut_class.value(data.mut_class),
-	abers.mut_type.value(data.mut_type),
-	abers.protein_seq_change.value(data.protein_seq_change),
-	abers.aber_user_id.value(data.user_id)).returning(abers.u_id);
-
-    Database.execute(aberInsertQuery, function(err, subresult) {
-	handleErr(err, subresult, aberInsertQuery);
-	var aber_u_id = subresult.rows[0].u_id;
-	data.aber_id = aber_u_id;
-	console.log(data);
-	exports.upsertSourceAnno(data, callback);
-    });
+	abers.gene_id.value(data.gene),
+	abers.mutation_class.value(data.mut_class),
+	abers.mutation_type.value(data.mut_type),
+	abers.locus.value(locus),
+	abers.original_amino_acid.value(acids[0]),
+	abers.new_amino_acid.value(acids[1]),
+	abers.last_edited.value(now),
+	abers.created_on.value(now)).returning(abers.id);
+
+	DjangoDatabase.execute(aberInsertQuery, function(err, subresult) {
+	    handleErr(err, subresult, aberInsertQuery);
+	    if (!err) {
+		var aber_u_id = subresult.rows[0].id;
+		data.aber_id = aber_u_id;
+		exports.upsertSourceAnno(data, callback);
+	    }
+	});
 }
 
-// note: requires an aber_id field to identify which aberration this source attaches to
-exports.upsertSourceAnno = function(data, callback) {
-    var sources = Schemas.aber_sources;
+// callback supplies user data
+function getUserInfo(user, callback) {
```

Comment:
Unclear if this is used anymore, since it seems to redirect to the MAGI Annotations.